### PR TITLE
Refactor vlm embedding routine to use precomputed feature

### PR DESCRIPTION
**Pull request by JustinTong0323**
## Motivation
- This PR is a Supplement for https://github.com/sgl-project/sglang/pull/6136

- Refactor vlm embedding routine to use precomputed feature

- Add tests for passing pixel values and precomputed features to VLMs(Split from test_vlm_acc)

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: test/srt/test_vlm_input_format.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
+import json
```
Comment (mickqian) [inline test/srt/test_vlm_input_format.py:1]: nit: is it better with `test_vlm_input_method`? I'm not sure about this

File: test/srt/test_vlm_input_format.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        )
+        self.assertIn("taxi", output["text"].lower())
+
+    async def test_understands_precomputed_features(self):
+        req = self.get_completion_request()
+        processor_output = self.get_processor_output(req=req)
+        with torch.inference_mode():
+            precomputed_features = self.__class__.visual(processor_output)
+        output = await self.engine.async_generate(
+            input_ids=processor_output["input_ids"][0].detach().cpu().tolist(),
```
Comment (mickqian) [inline test/srt/test_vlm_input_format.py:107]: will it go through the processor, where ids are expanded again?

File: test/srt/test_vlm_input_format.py (hunk 1 above)
Comment (JustinTong0323) [reply test/srt/test_vlm_input_format.py:1]: emm I think the main difference here is that we pass image data in different formats, like precomputed feature/pixel values/...

File: test/srt/test_vlm_input_format.py (hunk 2 above)
Comment (JustinTong0323) [reply test/srt/test_vlm_input_format.py:107]: When precomputed feature is given, the image is not passed to the processor, so the image token would not be expanded again, referring:
https://github.com/sgl-project/sglang/blob/2e37fa07bad9fa3ad7f5c091d4d860c722e2114b/python/sglang/srt/managers/multimodal_processors/qwen_vl.py#L128-L136

Comment (mickqian) [reply test/srt/test_vlm_input_format.py:107]: For qwen-vl, the ids seem to be safe. However, for others, since image_ids exist in input_ids but `images` is none, unexpected errors might be possible.
IMO, we should bypass the `process_mm_data` call in this case, since no data needs to be **processed**. Could be done in the future.

**Review by mickqian — APPROVED** [review]

File: test/srt/test_vlm_input_format.py (hunk 2 above)
Comment (JustinTong0323) [reply test/srt/test_vlm_input_format.py:107]: Got you, add to https://github.com/sgl-project/sglang/issues/5964
